cosmosapi/links_test.go

```diff
@@ -34,3 +34,12 @@ func TestResourceTypeFromLink(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateLinksUrlSafe(t *testing.T) {
+	assert.Equal(t, "dbs/mydb/colls/mycoll/triggers/%0D", createTriggerLink("mydb", "mycoll", "\r"))
+	assert.Equal(t, "dbs/mydb/colls/%7B%27collection%27:%27mycoll%27%7D", createCollLink("mydb", "{'collection':'mycoll'}"))
```

Comment:
I really hope no one ever creates a collection with this name. I probably makes the whole Azure data center in question burn to grounds :grimacing: